components/raftstore/src/store/worker/pd.rs

```diff
@@ -528,8 +527,12 @@ where
         spawn_local(f);
     }
 
+    // Note: The parameter doesn't contain `self` because this function may
+    // be called in an asynchronous context.
     fn handle_ask_batch_split(
-        &self,
+        router: RaftRouter<EK, ER>,
```

Comment:
Why change it into a function?

Reply:
This avoids using self inside the async context.

Reply:
It's not an async function.

Reply:
but it may be called in other async context

Reply:
rust cannot infer lifetime for `Runner`